tests/pact/fqns.repl

```diff
@@ -77,3 +77,66 @@
 
 (expect "selects correct test" (modB.get-test) "hello")
 (commit-tx)
+
+;;
+;; Module redeploy name resolution
+;;
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+
+  (defcap test ()
+    (enforce false "boom"))
+
+  (defun f ()
+    (with-capability (test)
+      1))
+  )
+(commit-tx)
+
+(begin-tx)
+(namespace 'free)
+(module test-mod-redeploy-ref g
+  (defcap g () true)
+  (defcap test ()
+    true)
+  (defun f ()
+    (with-capability (free.test-mod-redeploy-ref.test)
+      1))
+  )
+(expect "Namre resolution should find new reference" 1 (f))
```

Comment:
```suggestion
(expect "Name resolution should find new reference" 1 (f))
```

Reply:
Just for sanity here, can we show that the previous call to `test` would explode prior to upgrade?